lib/src/facts/linux/virtualization_resolver.cc

```diff
@@ -25,6 +25,29 @@ namespace lth_file = leatherman::file_util;
 
 namespace facter { namespace facts { namespace linux {
 
+    string virtualization_resolver::get_cloud_provider(collection& facts)
+    {
+        // Check for Azure
+        std::string provider = get_azure(facts);
+
+        return provider;
+
+    }
+
+    string virtualization_resolver::get_azure(collection& facts)
+    {
+        std::string provider;
+        if ( boost::filesystem::exists( "/var/lib/dhcp/dhclient.eth0.leases" ) )
+        {
+            string virt_what = agent::which("cat /var/lib/dhcp/dhclient.eth0.leases | grep option 245");
+            each_line(virt_what, [&](string& line) {
```

Comment:
What happens when the grep fails and virt_what is nil? Will the entire block be skipped? And then is it okay to return a null string provider?